code/mob/living/critter/ai.dm

```diff
@@ -217,20 +228,51 @@ var/list/ai_move_scheduled = list()
 		holder = null
 		..()
 
-	///Called when the task is switched to by the holder
+	/// Called when the task is switched to by the holder
 	proc/switched_to()
 
+	/// Called on every mobAI tick - this tick rate is determined by mobAI priority
 	proc/on_tick()
 
+	/// Called on every mobAI tick - returns a task to switch to, ending this task, or null to continue
 	proc/next_task()
 		return null
 
+	/// Called whenever the task is started or ended. Override this instead of reset()
 	proc/on_reset()
 		holder.target = null
 
-	proc/evaluate() // evaluate the current environment and assign priority to switching to this task
+	/// Evaluate the current environment and assign priority to switching to this task
+	proc/evaluate()
 		return 0
 
+	/// Returns a list of atoms that are potential targets for this task
+	proc/get_targets()
+		return list()
+
+	/// Takes a list of atoms which are then evaluated, before setting the holder's target. Note this checks a path exists to each target. The list of
+	/// targets is expected (but not required) to be ordered from best to worst - by default view() will do this if score_target() is based on distance
+	proc/get_best_target(list/atom/targets)
+		. = null
+		var/best_score = -INFINITY
+		var/list/best_path = null
+		if(length(targets))
+			for(var/atom/A in targets)
```

Comment:
do we need to be typechecking? we'd probably also want to check for ISQDELETED here

Reply:
it shouldn't need verification here unless someone does some weird stuff, since this is always called as `get_best_target(get_targets())`. So long as `get_targets()` returns valid results (and I would consider it a bug if it didn't), no additional typechecking is needed.

Reply:
then you can do an `as anything` to skip the `atom` typecheck